scheduler/core/components/optimizer/greedymax.py

```diff
@@ -41,97 +68,149 @@ def _allocate_time(plan: Plan, obs_time: timedelta) -> datetime:
 
         return start
 
-    def setup(self, selection: Selection) -> GreedyMaxOptimizer:
-        """
-        Preparation for the optimizer e.g. create chromosomes, etc.
-        """
-        self.groups = list(selection.schedulable_groups.values())
-        self.groupids = list(selection.schedulable_groups)
-        # self.scores = []
-        # for gid, group_data in selection.schedulable_groups.items():
-        #     self.groups.append(group_data)
-        period = len(list(selection.night_events.values())[0].time_grid)
-        print('Period: ', period)
-        self.timelines = [Timelines(selection.night_events, night) for night in range(period)]
+    def _find_max_group(self, plans: Plans):
+        """Find the group with the max score in an open interval"""
+
+        # the number of time slots in the minimum visit length
+        time_slot_length = plans.plans[self.sites[0]].time_slot_length
```

Comment:
That would be a bit more convenient.

Reply:
I'll include that in the PR above. 👍

Reply:
Since it is information that is being used from the `Collector`, I am hesitant to get at it in this way as it might not be a reliable way to achieve it. If I add it to the `Selection`, would that work for you?

Reply:
Yes, I think so.